packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/account-page-list.tsx

```diff
@@ -0,0 +1,327 @@
+import {
+  getProjection,
+  getTreeItemList,
+  isPageCollapsed,
+  getLastIndex,
+  TreeItem,
+} from "./utils";
+import { IDENTATION_WIDTH } from "./page-tree-item";
+import { PagesLoadingState } from "./pages-loading-state";
+
+type AccountPageListProps = {
+  accountId: string;
+  currentPageEntityId?: string;
+};
+
+const measuringConfig = {
+  droppable: {
+    strategy: MeasuringStrategy.Always,
+  },
+};
+
+export const AccountPageList: FunctionComponent<AccountPageListProps> = ({
+  currentPageEntityId,
+  accountId,
+}) => {
+  const { data, loading: pagesLoading } = useAccountPages(accountId);
+
+  const [createUntitledPage, { loading: createUntitledPageLoading }] =
+    useCreatePage(accountId);
+  const [createSubPage, { loading: createSubpageLoading }] =
+    useCreateSubPage(accountId);
+  const [reorderPage, { loading: reorderLoading }] = useReorderPage(accountId);
+  const [archivePage, { loading: archivePageLoading }] = useArchivePage();
+
+  const [initialLoading, setInitialLoading] = useState(true);
+
+  const loading =
+    pagesLoading ||
+    createUntitledPageLoading ||
+    createSubpageLoading ||
+    reorderLoading ||
+    archivePageLoading;
+
+  const [expandedPageIds, setExpandedPageIds] = useLocalstorageState<string[]>(
+    "hash-expanded-sidebar-pages",
+    [],
+  );
+
+  const [treeItems, setTreeItems] = useState<TreeItem[]>([]);
+
+  const [activeId, setActiveId] = useState<UniqueIdentifier | null>(null);
+  const [overId, setOverId] = useState<UniqueIdentifier | null>(null);
+  const [offsetLeft, setOffsetLeft] = useState(0);
+
+  // @todo handle loading/error states properly
+  const addPage = useCallback(async () => {
+    if (loading) {
+      return;
+    }
+
+    try {
+      await createUntitledPage(getLastIndex(treeItems));
+    } catch (err) {
+      // eslint-disable-next-line no-console -- TODO: consider using logger
+      console.error("Could not create page: ", err);
+    }
+  }, [createUntitledPage, loading, treeItems]);
+
+  const handleToggle = (nodeId: string) => {
+    setExpandedPageIds((expandedIds) =>
+      expandedIds.includes(nodeId)
+        ? expandedIds.filter((id) => id !== nodeId)
+        : [...expandedIds, nodeId],
+    );
+  };
+
+  const collapsedPageIds = useMemo(
+    () =>
+      treeItems
+        .filter((item) =>
+          isPageCollapsed(item, treeItems, expandedPageIds, activeId),
+        )
+        .map(({ page }) => page.entityId),
+    [treeItems, expandedPageIds, activeId],
+  );
+
+  const pagesFlatIdList = useMemo(
+    () => treeItems.map(({ page }) => page.entityId),
+    [treeItems],
+  );
+
+  useEffect(() => {
+    // We only update the state when everything has been loaded.
+    // If the request fails, pages will be reordered to their original state
+    if (!loading) {
+      setTreeItems(getTreeItemList(data));
+      setInitialLoading(false);
+    }
+  }, [data, loading]);
```

Comment:
This is not how to update derivied state when other state/props changes. https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes 

Let me know if this is confusing and we can chat about it

Reply:
Thank you @nathggns, I keep falling into this trap of using `useEffect`. Addressed this in f39a7ce8a6261d522bec2d935b1475e02c9ab675